matrix/tasks/health.py

```diff
@@ -53,4 +58,9 @@
     else:
         _log = rule.log.info
     _log("Health check: %s", result)
+
+    if result == 'unhealthy' and task.gating:
```

Comment:
If we do move `gating` up to the rule level, I think it would just be `rule.gating` without the additional assignment above.